src/adapters/IDB.js

```diff
@@ -329,8 +332,13 @@ export default class IDB extends BaseAdapter {
     return this.open().then(() => {
       return new Promise((resolve, reject) => {
         let results = [];
+        // Prepare filters
+        const remainingFilters = {...filters};
+        // If `indexField` was used already, don't filter again.
+        delete remainingFilters[indexField];
```

Comment:
I know you didn't write this code, but let's use the `omitKeys` function from the `utils` module here.

```js
const remainingFilters = omit(filters,  [indexKey]);
```